### PR TITLE
[FLINK-13818][web]: Check whether web submission are enabled

### PR DESCRIPTION
**Pull request by vthinkxie**
## What is the purpose of the change

fix https://issues.apache.org/jira/browse/FLINK-13818

## Brief change log

when `configuration.features['web-submit']` is false, hide the `Submit New Job` menu

## Verifying this change

If you put web.submit.enable: false into the configuration, the `Submit New Job` will hide in the menu

FireShot Capture 425 - Apache Flink Web Dashboard - localhost

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: flink-runtime-web/web-dashboard/src/app/interfaces/configuration.ts [hunk 1]
```diff
@@ -22,4 +22,7 @@ export interface ConfigurationInterface {
   'timezone-offset': number;
   'flink-version': string;
   'flink-revision': string;
+  features?: {
```
Comment (zentol) [inline flink-runtime-web/web-dashboard/src/app/interfaces/configuration.ts]: why are we checking this for null?

Comment (vthinkxie) [reply flink-runtime-web/web-dashboard/src/app/interfaces/configuration.ts]: Users may use this web for flink 1.9.0 and below, some company depends on flink ui to build there own flink platform.

**Comment by zentol** [comment]
No part of Flink provides any guarantees to work with older versions.
I'd rather not set such a precedent here for the WebUI without a dedicated discussions on the dev ML.

**Comment by vthinkxie** [comment]
We have given several public talks about how to integrating the flink web UI with a custom UI web, which encourages users including Flink web UI as a submodule in their project to get better user experience.
Ververica platform will also do this in the future version in the plan, I think we should not bring any break changes to the Rest API / Web UI without dedicated discussions on the dev ML.

**Comment by vthinkxie** [comment]
I don't think that adding this line of code means that we are guaranteeing the compatibility.
The only thing I do is try to stop making trouble for users and developers when possible.

`Changing this significantly alters how we have to approach new features, both for reviews and testing, and as such has to be discussed on the ML` is another thing IMO.

**Comment by zentol** [comment]
#7378 was about the  REST API, and ensures that existing applications working against the REST API continue to work. Since we provide backwards-compatibility guarantees for the REST API, this change was required.

As in, some application written against the 1.8 REST API continues to work in 1.9+.

What you are suggesting is to ensure the 1.10 WebUI continues to work against the 1.9- REST API, which is a _completely_ different thing.

**Comment by vthinkxie** [comment]
is there any discussion threads / ML that give the conclusion/decision that we must give backwards-compatibility guarantees for REST API, but must not give it to the Web?

**Comment by zentol** [comment]
>  is there any discussion threads / ML that give the conclusion/decision that we [...] must not give it to the Web(even when it is possible and easy to implement)?

No, because this topic has not come up in the past. Guess why I directed you _right away_ to the ML to start a discussion.

> [...] when we can make it work easily(with one line of code), why must not?

I've already stated my personal reason earlier:
> I'd rather not set such a precedent here for the WebUI[...]

In this very thread _you_ tried to use precedence as an argument for accepting this change, and I'm not going to single-handedly give people more ammo.

**Comment by vthinkxie** [comment]
I have removed the compatible code as you insist.
I don't think it is valuable enough to start a ML to discuss something about we should / should not trying best to keep compatibility in Web UI.

**Pull request merged**